plugins/inputs/win_wmi/win_wmi_test.go

```diff
@@ -0,0 +1,61 @@
+//go:build windows
+// +build windows
+
+package win_wmi
+
+import (
+	"fmt"
+	"os"
+	"regexp"
+	"testing"
+
+	"github.com/influxdata/telegraf/testutil"
+	"github.com/stretchr/testify/require"
+)
+
+// initialize test data
+var sysDrive = fmt.Sprintf(`%s\`, os.Getenv("SystemDrive")) // C:\
+var logger = new(testutil.Logger)
+var acc = new(testutil.Accumulator)
+
+// include Name as a tag, FreeSpace as a field, and Purpose as a known-null class property
+var testQuery Query = Query{
+	Namespace:            "ROOT\\cimv2",
+	ClassName:            "Win32_Volume",
+	Properties:           []string{"Name", "FreeSpace", "Purpose"},
+	Filter:               fmt.Sprintf(`NOT Name LIKE "\\\\?\\%%" AND Name LIKE "%s"`, regexp.QuoteMeta(sysDrive)),
+	TagPropertiesInclude: []string{"Name"},
+	tagFilter:            nil, // this is filled in by CompileInputs()
+}
+var expectedWql = fmt.Sprintf(
+	`SELECT Name, FreeSpace, Purpose FROM Win32_Volume WHERE NOT Name LIKE "\\\\?\\%%" AND Name LIKE "%s"`,
+	regexp.QuoteMeta(sysDrive))
+var testWmi Wmi = Wmi{Queries: []Query{testQuery}, Log: logger}
```

Comment:
Please declare the plugin in every test to avoid side effects between tests!

Reply:
Done. Thanks!